test/integration/SampleDataUtils.cpp

```diff
@@ -360,6 +373,69 @@ void SampleDataUtils::ifft(SampleDataUtils::CmplxArray& x)
     x /= x.size();
 }
 
+void SampleDataUtils::applyHannWindow(SampleDataUtils::CmplxArray& x)
+{
+    for (size_t t = 0; t < x.size(); ++t)
+    {
+        x[t] = x[t] * 0.5 * (1.0 - std::cos(2.0 * M_PI * t / x.size()));
+    }
+}
+
+std::vector<double> SampleDataUtils::powerSpectrum(const CmplxArray& fftVector)
+{
+    std::vector<double> spectrum;
+    spectrum.reserve(fftVector.size() / 2);
+    const double scale = 2.0 / (fftVector.size() * sqrt(2));
+    for (size_t i = 0; i < fftVector.size() / 2; ++i)
+    {
+        spectrum.push_back(std::abs(fftVector[i]) * scale);
+    }
+    return spectrum;
+}
+
+std::valarray<double> SampleDataUtils::powerSpectrumDB(const CmplxArray& fftVector)
+{
+    std::valarray<double> spectrum(fftVector.size() / 2);
+
+    const double scale = 2.0 / (fftVector.size() * sqrt(2));
+    for (size_t i = 0; i < fftVector.size() / 2; ++i)
+    {
+        spectrum[i] = 10 * log(std::abs(fftVector[i]) * scale);
+    }
+    return spectrum;
+}
+
+std::vector<std::pair<double, double>> SampleDataUtils::toPowerVector(std::valarray<double>& powerSpectrum,
+    size_t sampleRate)
+{
+    std::vector<std::pair<double, double>> v;
+    const double freqDelta = sampleRate / 2.0 / powerSpectrum.size();
+    double f = 0;
+    for (auto m : powerSpectrum)
+    {
+        v.push_back({f, m});
+        f += freqDelta;
+    }
+    return v;
+}
+
+std::vector<std::pair<double, double>> SampleDataUtils::isolatePeaks(std::vector<std::pair<double, double>>& powerFreq,
```

Comment:
Perhaps we will use this more in tests. It tuned out to be more efficient for c9 where the tone is poluted.